### PR TITLE
Preserve type bounds for inlined definitions in posttyper

### LINKED ISSUE #17168: Summon allows to ignore type bounds
## Compiler version

3.3.0-RC1
## Minimized code

```Scala
scala> type F[X <: String] = X
// defined alias type F[X <: String] = X
                                                                                
scala> val a = summon[F[Int] =:= Int]
val a: Int =:= Int = generalized constraint
```

## Output
Works. Both in REPL and Compiler.

## Expectation
```Scala
-- [E057] Type Mismatch Error: -------------------------------------------------
  |val a = summon[F[Int] =:= Int]
  |                 ^
  |                 Type argument Int does not conform to upper bound String
```

### PR DESCRIPTION
**Pull request by Decel**
Fixes #17168, as per @odersky idea.

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]

File: compiler/src/dotty/tools/dotc/transform/PostTyper.scala [hunk 1]
```diff
@@ -363,6 +363,7 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
         case Inlined(call, bindings, expansion) if !call.isEmpty =>
           val pos = call.sourcePos
           CrossVersionChecks.checkExperimentalRef(call.symbol, pos)
+          super.transform(call)
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/transform/PostTyper.scala:366]: We still have the issue of the exponential blowup.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PostTyper.scala:366]:
```suggestion
         transform(call)
```

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PostTyper.scala]: By doing this recursion we will always enable `ctx.compilationUnit.needsInlining = true`.

* https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L287
* https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L295
* https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L348

This is not ideal if the file only contains transparent inline calls.

**Review by nicolasstucki — CHANGES_REQUESTED** [review]
We have to be more careful with that recursion.

We already know about 2 unintended consequences, and we may be missing something else.

**Comment by odersky** [comment]
Yes, let's back out of the last commit.

**Comment by dwijnand** [comment]
The last commit has the wrong indentation.

**Review by nicolasstucki — COMMENTED** [review]
@Decel commit LGTM. 

@dwijnand could you review the commit I added?

**Review by dwijnand — APPROVED** [review]

**Pull request merged**